src/Microsoft.Diagnostics.Monitoring/Contracts/IDiagnosticServices.cs

```diff
@@ -28,21 +37,18 @@ public enum DumpType
         Triage
     }
 
-    public enum TraceState
-    {
-        Stopped = 0,
-        Running = 1,
-    }
-
     public sealed class TraceRequest
     {
         public string Configuraton { get; set; }
     }
 
-    public sealed class OperationResult<T>
+    public class OperationResult
```

Comment:
Note that pid is optional for all these apis now. If we don't return it in some way, then there is no way for the caller to know what the pid is. I am also expecting that we use it in the future as we tighten up the spec. For example, the current {guid}.nettrace should probably be changed to rawLogs_<pid/uuid>.nettrace.

Reply:
If we want the Pid we could always have an API on IDiagnosticServices:
````
int GetPid(ProcessReference proc_reference)
````
That seems simple and more flexible than creating nested generic types to aggregate all the information into the return value from a single call.

Reply:
I'd suggest doing the resolution once up-front rather than embedding an optional resolution step in every IDiagnosticServices API:
````
if(pid == null)
{
    pid = diagnosticServices.ResolveCurrentDefaultProcess();
}
IStreamResult streamResult = diagnosticServices.GetTrace(pid);
````